metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -220,13 +220,24 @@ private boolean isValidNewLeader(int replica) {
     private void tryElection(PartitionChangeRecord record) {
         ElectionResult electionResult = electLeader();
         if (electionResult.node != partition.leader) {
-            log.debug(
-                "Setting new leader for topicId {}, partition {} to {} using {} election",
-                topicId,
-                partitionId,
-                electionResult.node,
-                electionResult.unclean ? "an unclean" : "a clean"
-            );
+            // generating log messages for partition elections can get expensive on large clusters,
+            // so only log clean elections at TRACE level;
+            // log unclean elections at WARN level since doing so can lead to data loss.
+            if (electionResult.unclean) {
+                log.warn(
```

Comment:
How about INFO level? Unclean election "expected behavior" if it is enabled or requested specifically.

Reply:
Sure, sounds good.  Will adjust.